alacritty_terminal/src/renderer/rects.rs

```diff
@@ -26,24 +26,20 @@ pub struct Rect<T> {
     pub y: T,
     pub width: T,
     pub height: T,
+    pub color: Rgb
 }
 
 impl<T> Rect<T> {
-    pub fn new(x: T, y: T, width: T, height: T) -> Self {
-        Rect { x, y, width, height }
+    pub fn new(x: T, y: T, width: T, height: T, color: Rgb) -> Self {
+        Rect { x, y, width, height, color}
     }
 }
 
-struct Line {
-    rect: Rect<f32>,
-    start: Point,
-    color: Rgb,
-}
-
-impl Line {
-    /// Create a line that starts on the left of `cell` and is one cell wide
-    fn from_cell(cell: &RenderableCell, flag: Flags, metrics: &Metrics, size: &SizeInfo) -> Line {
-        let cell_x = cell.column.0 as f32 * size.cell_width;
+impl Rect<f32> {
+    fn from_line(line: &Line, flag: Flags, metrics: &Metrics, size: &SizeInfo, color: Rgb) -> Rect<f32> {
```

Comment:
I think moving this to `Line` and calling it as `into_rect` is better since we can then use `&self`.